pkg/cloudprovider/events/events.go

```diff
@@ -38,3 +42,13 @@ func NodeClaimFailedToResolveNodeClass(nodeClaim *v1beta1.NodeClaim) events.Even
 		DedupeValues:   []string{string(nodeClaim.UID)},
 	}
 }
+
+func FilteredInstanceTypes(nodeClaim *v1beta1.NodeClaim, filteredITsRequirement int, filteredExoticITs int, filteredExpensiveSpot int, filteredITs []string) events.Event {
+	return events.Event{
+		InvolvedObject: nodeClaim,
+		Type:           v1.EventTypeNormal,
+		Message: fmt.Sprintf("Filtered %d instance types on requirements, filtered %d instance types from exotic instance types, filtered %d instance types with spot offerings more expensive than cheapest on-demand. Remaining instance types:: %s",
+			filteredITsRequirement, filteredExoticITs, filteredExpensiveSpot, utils.PrettySlice(filteredITs, 5)),
```

Comment:
+1 to this. If we can build this so that it only adds the "filtered" statement when it actually filters something out, it's going to make this statement a lot more concise